### PR TITLE
Update the naming around "archive" and "delete" verbs in mmctl

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

This PRs adds new archive commands in order to differentiate the soft and hard deletion commands. Also it includes a refactor for the "team" command that moves the "team user" logic to a sub command "users".

I've kept the old "remove*"commands and added a deprecation message to warning the users to start avoiding this command

I left these two commands untouched because I thing the name of the command fits with the logic:

- license::removeLicenseCmdF: Clean the license but doesn't remove inserts or updates existing data.
- permissions::removePermissionsCmdF: Patches de role removing permissions

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-19200

### CODE REVIEW

File: commands/team.go [hunk 1]
```diff
@@ -29,28 +29,13 @@ var TeamCreateCmd = &cobra.Command{
...
-	RunE:    withClient(addUsersCmdF),
-}
-
 var DeleteTeamsCmd = &cobra.Command{
 	Use:   "delete [teams]",
 	Short: "Delete teams",
 	Long: `Permanently delete some teams.
 Permanently deletes a team along with all related information including posts from the database.`,
 	Example: "  team delete myteam",
+	Args:    cobra.MinimumNArgs(1),
```
Comment (mgdelacroix) [inline commands/team.go]: ❤️

File: commands/team_users.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var TeamUsersCmd = &cobra.Command{
+	Use:   "users",
+	Short: "Management of team users",
+}
+
+var RemoveUsersCmd = &cobra.Command{
```
Comment (mgdelacroix) [inline commands/team_users.go]: To keep consistent with the naming schema, we should use:

```suggestion
var TeamUsersRemoveCmd = &cobra.Command{
```

and same with the command function name (`teamUsersArchiveCmdF`), test functions, etc.

File: commands/team_users.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	Use:        "remove [team] [users]",
+	Short:      "Remove users from team",
+	Long:       "Remove some users from team",
+	Example:    "  team remove myteam user@example.com username",
+	Deprecated: "please use \"archive\" instead",
+	Args:       cobra.MinimumNArgs(2),
+	RunE:       withClient(archiveUsersCmdF),
+}
+
+var ArchiveUsersCmd = &cobra.Command{
```
Comment (mgdelacroix) [inline commands/team_users.go]: Same here:

```suggestion
var TeamUsersArchiveCmd = &cobra.Command{
```

File: commands/team_users.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	Use:     "archive [team] [users]",
+	Short:   "Archive users from team",
+	Long:    "Archive some users from team",
+	Example: "  team archive myteam user@example.com username",
+	Args:    cobra.MinimumNArgs(2),
+	RunE:    withClient(archiveUsersCmdF),
+}
+
+var AddUsersCmd = &cobra.Command{
```
Comment (mgdelacroix) [inline commands/team_users.go]:
```suggestion
var TeamUsersAddCmd = &cobra.Command{
```

File: commands/team_users.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	RunE:       withClient(archiveUsersCmdF),
+}
+
+var TeamUsersArchiveCmd = &cobra.Command{
+	Use:     "archive [team] [users]",
+	Short:   "Archive users from team",
+	Long:    "Archive some users from team",
+	Example: "  team archive myteam user@example.com username",
+	Args:    cobra.MinimumNArgs(2),
+	RunE:    withClient(archiveUsersCmdF),
```
Comment (mgdelacroix) [inline commands/team_users.go]: We should modify the function names as well:

```suggestion
	RunE:    withClient(teamUsersArchiveCmdF),
```

File: commands/team_users_test.go [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+// See LICENSE.txt for license information.
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mmctl/printer"
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlUnitTestSuite) TestArchiveUserCmd() {
```
Comment (mgdelacroix) [inline commands/team_users_test.go]: And their correspondant test functions:

```suggestion
func (s *MmctlUnitTestSuite) TestTeamUsersArchiveCmd() {
```

Comment (ethervoid) [reply commands/team_users_test.go]: Uops sorry

**Comment by wiersgallak** [comment]
I chatted with @mgdelacroix on this for my understanding, but want to ensure I have it documented on the PR for documenting in our more formal documentation and to ensure I am accurate in what the experience will be: 
 - When running a command that has been deprecated, the command will print the deprecation statement.  For instance, if a user types ``command delete commandID`` the output will be ``please use \"archive\" instead``
 - When asking for the help menu, a user will not see any command that is deprecated.

File: commands/team_users.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	Use:   "users",
+	Short: "Management of team users",
+}
+
+var RemoveUsersCmd = &cobra.Command{
+	Use:        "remove [team] [users]",
+	Short:      "Remove users from team",
+	Long:       "Remove some users from team",
+	Example:    "  team remove myteam user@example.com username",
+	Deprecated: "please use \"archive\" instead",
```
Comment (wiersgallak) [inline commands/team_users.go]: I am not sure that archiving users from a team is an obvious command.  We do not use this phrasing anywhere else in the product.  I would recommend not using the archive command/not deprecating the ``remove users from team``,  as remove is more obvious that you are removing one or more users from their access to the team.

Comment (ethervoid) [reply commands/team_users.go]: Good for me, changing

**Comment by ethervoid** [comment]
@wiersgallak :

> When running a command that has been deprecated, the command will print the deprecation statement. For instance, if a user types command delete commandID the output will be please use \"archive\" instead

Yes, the user will get the message: `Command "delete" is deprecated, please use "archive" instead
`

> When asking for the help menu, a user will not see any command that is deprecated.

Yes, that's correct. Those commands are hidden even though you can still use them, for example if you're calling those commands from a script

**Comment by ethervoid** [comment]
@wiersgallak Let me know what this PR is ok for you to merge :)

File: docs/mmctl_command_archive.md [hunk 8]
```diff
@@ -1,25 +1,25 @@
...
+Archive a slash command
 
 ### Synopsis
 
-Delete a slash command. Commands can be specified by command ID.
+Archive a slash command. Commands can be specified by command ID.
 
 ```
-mmctl command delete [flags]
+mmctl command archive [flags]
```
Comment (wiggin77) [inline docs/mmctl_command_archive.md]: I know this is already merged, but for consistency with the other commands shouldn't this be 
```
mmctl command archive commandID [flags]
```
?  @ethervoid   @mgdelacroix

File: commands/command.go [hunk 9]
```diff
@@ -36,12 +36,22 @@ var CommandListCmd = &cobra.Command{
...
+	Short:      "Delete a slash command",
+	Long:       `Delete a slash command. Commands can be specified by command ID.`,
+	Example:    `  command delete commandID`,
+	Deprecated: "please use \"archive\" instead",
+	Args:       cobra.ExactArgs(1),
+	RunE:       withClient(archiveCommandCmdF),
+}
+
+var CommandArchiveCmd = &cobra.Command{
+	Use:     "archive",
```
Comment (wiggin77) [inline commands/command.go]: In most of the other CLI docs, when a option requires a non-flag argument it is listed  i.e.  ```
var CommandArchiveCmd = &cobra.Command{
	Use:     "archive [commandID]",
```
There are a few in this file that do no conform.  @ethervoid   @mgdelacroix

File: commands/command.go [hunk 10]
```diff
@@ -160,7 +171,7 @@ func listCommandCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func deleteCommandCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+func archiveCommandCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	ok, response := c.DeleteCommand(args[0])
 	if response.Error != nil {
 		return errors.New("Unable to delete command '" + args[0] + "' error: " + response.Error.Error())
```
Comment (wiggin77) [inline commands/command.go]: Should this error message be changed to "Unable to archive command..." ?
@ethervoid   @mgdelacroix

**Comment by ethervoid** [comment]
@wiggin77 thank you for you review :) . I've created a [new PR](https://github.com/mattermost/mmctl/pull/108) with those issues fixed

**Pull request merged**